mkosi/machine.py

```diff
@@ -178,6 +184,9 @@ def setUpClass(cls) -> None:
         cls.machine.build()
 
     def setUp(self) -> None:
+        # Replacing underscores which makes name invalid.
+        # Necessary for shell otherwise racing conditions to the disk image will happen.
+        self.machine.args.hostname = self._testMethodName.replace("_", "-")
```

Comment:
_testMethodName is an internal method, why not use the `id()' method here? https://docs.python.org/3/library/unittest.html#unittest.TestCase.id

Reply:
Just because it'd look better than` self.id().split(".")[3]` to get the name. But this is a bad practice, right? so maybe I should do it this way.